src/test/java/it/pagopa/pn/bff/service/TosPrivacyServiceTest.java

```diff
@@ -41,6 +41,101 @@ public static void setup() {
         tosPrivacyService = new TosPrivacyService(pnUserAttributesClient, pnBffExceptionUtility);
     }
 
+    @Test
+    void getPgConsents() {
+        Consent tosConsent = consentsMock.getPgTosConsentResponseMock();
+        List<it.pagopa.pn.bff.generated.openapi.server.v1.dto.ConsentType> type = new ArrayList<>();
+        type.add(it.pagopa.pn.bff.generated.openapi.server.v1.dto.ConsentType.TOS_DEST_B2B);
+
+        when(pnUserAttributesClient.getPgConsentByType(
+                Mockito.anyString(),
+                Mockito.any(CxTypeAuthFleet.class),
+                Mockito.any(ConsentType.class)
+        ))
+                .thenReturn(Mono.just(tosConsent));
+
+        StepVerifier.create(tosPrivacyService.getPgTosPrivacy(
+                        UserMock.PN_UID,
+                        it.pagopa.pn.bff.generated.openapi.server.v1.dto.CxTypeAuthFleet.PG,
+                        type
+                ))
+                .expectNextSequence(consentsMock.getBffTosPrivacyConsentMock())
```

Comment:
On the when condition above you are telling that the micro-service api returns `tosConsent` (correct), but here you are checking that the service returns `consentsMock.getBffTosPrivacyConsentMock()` (wrong).

Reply:
done